File: typescript/packages/management/src/tools/connect_linear.ts
import type { ToolContext, ToolHandler } from '../registrar.js';

type ConnectLinearRequest = Parameters<
  ToolContext['client']['sources']['connectLinear']
>[1];

export const connectLinear: ToolHandler = async (
  ctx: ToolContext,
  input: Record<string, unknown>,
) => {
  const request: ConnectLinearRequest = {
    api_key: String(input.api_key),
  };
  if (typeof input.name === 'string') {
    request.name = input.name;
  }
  if (Array.isArray(input.include)) {
    request.include = input.include.map(String) as ConnectLinearRequest['include'];
  }
  if (typeof input.rate_cap_per_minute === 'number') {
    request.rate_cap_per_minute = input.rate_cap_per_minute;
  }
  if (typeof input.poll_interval_seconds === 'number') {
    request.poll_interval_seconds = input.poll_interval_seconds;
  }
  return ctx.client.sources.connectLinear(ctx.sandboxId, request);
};

File: python/copass-management/src/copass_management/tools/connect_linear.py
from __future__ import annotations

from typing import Any, Dict, TYPE_CHECKING

if TYPE_CHECKING:
    from copass_management.registrar import ToolContext


async def connect_linear(ctx: "ToolContext", input: Dict[str, Any]) -> Any:
    kwargs: Dict[str, Any] = {
        "api_key": str(input["api_key"]),
    }
    if input.get("name") is not None:
        kwargs["name"] = str(input["name"])
    if isinstance(input.get("include"), list):
        kwargs["include"] = [str(x) for x in input["include"]]
    if input.get("rate_cap_per_minute") is not None:
        kwargs["rate_cap_per_minute"] = int(input["rate_cap_per_minute"])
    if input.get("poll_interval_seconds") is not None:
        kwargs["poll_interval_seconds"] = int(input["poll_interval_seconds"])
    return await ctx.client.sources.connect_linear(ctx.sandbox_id, **kwargs)
